src/neuroglancer/ui/tool.ts

```diff
@@ -287,11 +297,12 @@ export class ToolBinder extends RefCounted {
     super.disposed();
   }
 
-  private deactivate() {
+  deactivate() {
```

Comment:
I think this should remain private, and instead you could add a `cancel` method to `ToolActivation`.

Reply:
I am not sure how to do that while preserving the ideal behavior. For tools to be able to deactivate themselves, it seems like they need to be setting `activeTool` in `ToolBinder` to `undefined`. I can add another public method to `ToolBinder` which just calls `deactivate` and leave `deactivate` private, but that would just be redundant from making `deactivate` itself public. Is there another strategy here that I'm not seeing?

Reply:
My thinking was just that by making the "public" interface a method of ToolActivation, it is clearer that it is intended only for a tool to deactivate itself, and also allows you to check that the tool is actually active, so that it isn't deactivated by mistake if event handlers/signals are run in an unexpected order.

You raise a good point that some public method of `ToolBinder` is needed --- unfortunately the public/private access control provided by typescript is not quite adequate for this case.  I would suggest renaming `deactivate` to `deactivate_` and making it public, but indicating it is for internal use only.  Then you can call it from a `cancel` method of ToolActivation.

Reply:
That makes sense. The only issue I can foresee is that in our branch, we have extra controls in the sidebar for multicut submit/cancel (Chris added these but says he is not opposed to removing them). If we keep those, they need to be able to deactivate the tool on submit or cancel, but they don't have a reference to the `ToolActivation` (since the tool was created with `makeToolButton`, and these buttons are outside the tool). They do have a reference to the layer, so they can call `layer.manager.root.toolBinder.deactivate_()`, but if that's an "internal" method then they probably should not be calling that. Do you have any thoughts on this case? In general it seems like it might be useful to deactivate the current tool from elsewhere in the code.

Reply:
I think the multicut tool, like all tools when activated, should open a status area at the bottom that shows the relevant key bindings and current multicut state.  I think that would also be a good place to put any additional buttons for cancel/submit, and you will have access to the ToolActivation there as well.

Reply:
Okay, the team agrees that is a good idea (in fact we already had buttons there for cancel/submit; the sidebar buttons were extra but we can just remove those). I pushed a change to support this.